### DISCUSSION TITLE
hurl package not supporting the Docker build command

### DISCUSSION THREAD

**Discussion by fev1n** [Orange-OpenSource/hurl#1823] [Q&A] [answered]
Is hurl package incompatible with MacOS having apple silicone chips? 

I am making one project where I need to use hurl for integration testing, this project already uses docker in it. Before all the functionalities for docker were working but after getting hurl into in my project using Node, I encountered issue which makes no sense to me. 

Here is a screenshot of error I face, after running ```docker compose up```:

<img width="689" alt="image" src="https://github.com/Orange-OpenSource/hurl/assets/132969492/0801a83a-c32d-48bb-9a03-2ac64faa6a2c">

**Comment by jcamiel** [comment, marked as answer]
Hi @fev1n could you share your complete Dockerfile so I can try to understand and reproduce your issue?

I've the impression that you're installing Hurl as a npm dependencies. As this installation is inside a Linux container, on an ARM platform (Apple M1/2), the Hurl node package try to install the Linux ARM variant (which we don't provide BTW). One thing you can test is to install Hurl as a native binary in your Docker. For instance, if your Docker base image is a Debian, you can install like this <https://hurl.dev/docs/installation.html#debian-ubuntu>. Depending on your base image, Hurl installation can vary. Then, in your `package.json`, you can call it with `"test": "hurl *.hurl"` instead of `"test": "npx hurl *.hurl"` (and you should remove the Hurl dependencies from your dev dependencies).

This way, there'll be less layer and it will be easier to debug. With your Dockerfile, I can try to test it.

You can check also our tutorial on CI/CD integration, that may help you => https://hurl.dev/docs/tutorial/ci-cd-integration.html
